File: app/src/main/java/com/icerockdev/babenko/core/dagger/modules/ContextModule.java
package com.icerockdev.babenko.core.dagger.modules;

/**
 * Created by Roman Babenko on 04/06/17.
 */

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {
    private Context mContext;

    public ContextModule(Context context) {
        mContext = context;
    }

    @Singleton
    @Provides
    Context getContext() {
        return mContext;
    }
}

File: app/src/main/java/com/icerockdev/babenko/core/dagger/modules/ContextModule.kt
package com.icerockdev.babenko.core.dagger.modules

/**
 * Created by Roman Babenko on 04/06/17.
 */

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ContextModule(@get:Singleton
                    @get:Provides
                    internal val context: Context)
